src/nodes/accessors/InstanceNode.js

```diff
@@ -265,6 +280,81 @@ class InstanceNode extends Node {
 
 		}
 
+		if ( this.previousInstanceMatrixNode !== null ) {
+
+			frame.object.previousInstanceMatrix.array.set( this.instanceMatrix.array );
+
+		}
+
+	}
+
+	/**
+	 * Computes the transformed/instanced vertex position of the previous frame.
+	 *
+	 * @param {NodeBuilder} builder - The current node builder.
+	 * @return {Node<vec3>} The instanced position from the previous frame.
+	 */
+	getPreviousInstancedPosition( builder ) {
+
+		const instancedMesh = builder.object;
+
+		if ( this.previousInstanceMatrixNode === null ) {
+
+			instancedMesh.previousInstanceMatrix = this.instanceMatrix.clone();
+
+			const { instanceMatrix } = this;
+			const { count } = instanceMatrix;
+
+			if ( this.isStorageMatrix ) {
+
+				this.previousInstanceMatrixNode = storage( instanceMatrix, 'mat4', Math.max( count, 1 ) ).element( instanceIndex );
+
+			} else {
+
+				// Both backends have ~64kb UBO limit; fallback to attributes above 1000 matrices.
+
+				if ( count <= 1000 ) {
+
+					this.previousInstanceMatrixNode = buffer( instanceMatrix.array, 'mat4', Math.max( count, 1 ) ).element( instanceIndex );
+
+				} else {
+
+					const interleaved = new InstancedInterleavedBuffer( instanceMatrix.array, 16, 1 );
+
+					const bufferFn = instanceMatrix.usage === DynamicDrawUsage ? instancedDynamicBufferAttribute : instancedBufferAttribute;
+
+					const instanceBuffers = [
+						bufferFn( interleaved, 'vec4', 16, 0 ),
+						bufferFn( interleaved, 'vec4', 16, 4 ),
+						bufferFn( interleaved, 'vec4', 16, 8 ),
+						bufferFn( interleaved, 'vec4', 16, 12 )
+					];
+
+					this.previousInstanceMatrixNode = mat4( ...instanceBuffers );
+
+				}
+
+			}
+
+		}
+
+		return this.previousInstanceMatrixNode.mul( positionPrevious ).xyz;
+
+	}
+
+	/**
+	 * Returns `true` if instance matrices from the previous frame are required. Relevant
+	 * when computing motion vectors with {@link VelocityNode}.
+	 *
+	 * @param {NodeBuilder} builder - The current node builder.
+	 * @return {boolean} Whether instance matrices from the previous frame are required or not.
+	 */
+	needsPreviousInstanceMatrices( builder ) {
```

Comment:
We need the exact same helper in `InstanceNode` and `SkinningNode`. Could we move this to `Node` and rename it to `needsPreviousData()`?